crates/bevy_pbr/src/render/light.rs

```diff
@@ -600,6 +599,12 @@ pub fn prepare_lights(
 
     let mut point_lights: Vec<_> = point_lights.iter().collect::<Vec<_>>();
 
+    let nb_shadow_lights = point_lights
+        .iter()
+        .filter(|light| light.1.shadows_enabled)
+        .count()
+        .min(MAX_POINT_LIGHT_SHADOW_MAPS);
```

Comment:
We have to also respect the max texture array layers limit here:
```suggestion
    let nb_shadow_lights = point_lights
        .iter()
        .filter(|light| light.1.shadows_enabled)
        .count()
        .min(MAX_POINT_LIGHT_SHADOW_MAPS)
        .min(render_device.limits().max_texture_array_layers / 6);
```